src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -829,9 +852,18 @@ private static VariantContext phaseVC(final VariantContext vc, final String ID,
         final List<Genotype> phasedGenotypes = new ArrayList<>();
         for ( final Genotype g : vc.getGenotypes() ) {
             final List<Allele> alleles = g.getAlleles();
-            if (phaseGT.equals(phase10) && g.isHet()) Collections.reverse(alleles); // swap the alleles if heterozygous
+            final List<Allele> newAlleles = new ArrayList<>(alleles);
+            if (phaseGT.equals(phase10) && g.isHet()) {
```

Comment:
I think this series of nested conditionals could be simplified as:
```
final int phasedFirstAlleleIndex = phaseGT.equals(phase10) ? 0 : 1;
if (g.isHet() && ! isSiteSpecificAltAllele(newAlleles.get(phasedFirstAlleleIndex))) {
    Collections.reverse(newAlleles);
}
```

I would also consider making the phase GTs into an `enum` instead of hardcoded strings, and giving it a method to extract which index, 0 or 1, is first.

Reply:
Good suggestions, I've implemented these changes.